mule-extensions-api/src/main/java/org/mule/runtime/extension/api/client/DefaultOperationParameters.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.client;
+
+import static com.google.common.collect.ImmutableMap.copyOf;
+import static java.util.Collections.emptyMap;
+import static java.util.Optional.*;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Default {@link OperationParameters} implementation, works as a generic {@link OperationParameters} for every extension operation.
+ *
+ * @since 1.0
+ */
+public class DefaultOperationParameters implements OperationParameters {
+
+  private final String configRef;
+  private final Map<String, Object> parameters;
+
+  DefaultOperationParameters(String configRef, Map<String, Object> parameters) {
+    this.configRef = configRef;
+    this.parameters = copyOf(parameters);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Optional<String> getConfigName() {
+    return ofNullable(configRef);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<String, Object> getParameters() {
+    return copyOf(parameters);
+  }
+
+  /**
+   * @return a new {@link DefaultOperationParameters.Builder} instance to create parameters to execute an extension operation
+   * using the {@link ExtensionsClient}.
+   */
+  public static Builder builder() {
```

Comment:
this static method should be on the interface. Usage of this should be `OperationParameters.builder().bleh()`